core/domain/feedback_services.py

```diff
@@ -366,6 +366,34 @@ def _get_thread_from_model(thread_model):
         thread_model.created_on, thread_model.last_updated)
 
 
+def get_messages_for_exp_id(exp_id, limit=None):
```

Comment:
Thinking about this again - I agree that increasing the datastore queries is too big a compromise for this. Thus I've followed your original suggestion - removed the limit parameter from the feedback_services method, created a new method for the model's API and used it to fetch the latest 3 feedback threads and finally extracted the most recent message out of them and sent this to the frontend - so it's now a lot less no. of queries to the datastore!